tellerlogger/src/main/java/com/kiliansteenman/teller/logger/ui/detail/TellerDetailViewModel.kt

```diff
@@ -0,0 +1,58 @@
+package com.kiliansteenman.teller.logger.ui.detail
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.viewModelScope
+import androidx.room.Room
+import com.kiliansteenman.teller.logger.data.TellerLog
+import com.kiliansteenman.teller.logger.data.room.TellerLogDatabase
+import com.kiliansteenman.teller.logger.share.toSharableContent
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
+
+internal class TellerDetailViewModel(
+    application: Application,
+    savedStateHandle: SavedStateHandle,
+) : AndroidViewModel(application) {
+
+    private val db: TellerLogDatabase
+        get() {
+            return Room.databaseBuilder(
+                getApplication(),
+                TellerLogDatabase::class.java, "database-name"
+            ).build()
+        }
+
+    private val logId = savedStateHandle.get<Long>(PARAM_LOG_ID)
+        ?: throw IllegalArgumentException("Log id argument should not be null")
+
+    private val _events = MutableSharedFlow<TellerEvent>(0, extraBufferCapacity = Integer.MAX_VALUE)
+    val events: Flow<TellerEvent> = _events
+
+    private val _logEvent = MutableLiveData<TellerLog>()
```

Comment:
I think it makes sense to convert this to a flow while we are at it so we can get rid of the LiveData dependency.